pySearch.py

```diff
@@ -48,13 +48,14 @@ def openBrowser(self):
     #end of openBrowser()
 #end of Query
 
-searchObj = Search(args.s)
 if args.engine is not None:
     searchObj.setEngine(args.engine[0])
 
 if args.domain is not None:
     searchObj.setdomain(args.domain[0])
 #end of cmd args handling
 
-searchObj.buildLink()
-searchObj.openBrowser()
+for search in args.s:
+    searchObj = Search(search)
```

Comment:
Moving this down from line 50 or 51 means the setters are run on a nonexistent object. I'm surprised Python let this slide. This means the command line argument setters for `engine` and `domain` are essentially ignored. Everything works right now as the defaults are being used by the `Search` objects created in this loop.

You can fix this by creating an empty `Search` object on line 51 so nothing is passed into the initial `Search` object. Then create a setter for `searchQuery` and use the setter to change the search query during each loop. This way you avoid creating a new `Search` object every time and the engine and domain settings are preserved.

Also, an odd behavior that I ran into is, if you completely close your browser and try to execute multiple searches a fast machine will open each search in a new window. On a slower machine like my OSX vm it will open up each one search in a new window and the others in another window. I don't know if this is reproducible for others but I think it's something we should take a look at. If this behavior is reproducible I think it can be fixed by adding a mild delay to the end of the for loop so the initial browser open has time to complete.